#
# clamav-0.95.2 is the latest clamav version. However indimail-1.5.5 details
# installation of clamav-0.95.1
# Apply this if you want to install clamav-0.95.2 instead of clamav-0.95.1 
#
diff -Naur Clamav-0.95.2/docs/man/freshclam.1.in clamav-0.95.2/docs/man/freshclam.1.in
--- Clamav-0.95.2/docs/man/freshclam.1.in	2009-03-17 00:07:26.000000000 +0530
+++ clamav-0.95.2/docs/man/freshclam.1.in	2009-06-15 14:28:58.000000000 +0530
@@ -53,6 +53,10 @@
 \fB\-\-daemon\-notify=/path/to/clamd.conf\fR
 Notify the daemon about the new database. By default it reads a hardcoded config file but you can use a different one. Both local and TCP sockets are supported.
 .TP 
+\fB\-f, \-\-foreground\fR
+Do not detach to the background if \-d option has been specfied. Useful when configured under
+.BR supervise(8) .
+.TP 
 \fB\-a IP, \-\-local\-address=IP\fR
 Use (local) IP for HTTP downloads. Useful for multi\-homed systems. If binding fails for whatever reason, a warning is issued and freshclam behaves like without this flag.
 .TP 
diff -Naur Clamav-0.95.2/etc/clamd.conf clamav-0.95.2/etc/clamd.conf
--- Clamav-0.95.2/etc/clamd.conf	2009-04-07 16:02:06.000000000 +0530
+++ clamav-0.95.2/etc/clamd.conf	2009-06-15 14:30:11.000000000 +0530
@@ -62,7 +62,7 @@
 
 # Path to the database directory.
 # Default: hardcoded (depends on installation options)
-#DatabaseDirectory /var/lib/clamav
+#DatabaseDirectory /var/indimail/share/clamd
 
 # The daemon can work in local mode, network mode or both. 
 # Due to security reasons we recommend the local mode.
@@ -167,7 +167,7 @@
 
 # Run as another user (clamd must be started by root for this option to work)
 # Default: don't drop privileges
-#User clamav
+#User qscand
 
 # Initialize supplementary group access (clamd must be started by root).
 # Default: no
@@ -233,7 +233,7 @@
 # With this option clamav will try to detect broken executables (both PE and
 # ELF) and mark them as Broken.Executable.
 # Default: no
-#DetectBrokenExecutables yes
+DetectBrokenExecutables yes
 
 
 ##
diff -Naur Clamav-0.95.2/freshclam/freshclam.c clamav-0.95.2/freshclam/freshclam.c
--- Clamav-0.95.2/freshclam/freshclam.c	2009-05-04 14:16:20.000000000 +0530
+++ clamav-0.95.2/freshclam/freshclam.c	2009-06-15 14:30:50.000000000 +0530
@@ -132,6 +132,7 @@
     mprintf("    --config-file=FILE                   read configuration from FILE.\n");
     mprintf("    --log=FILE           -l FILE         log into FILE\n");
     mprintf("    --daemon             -d              run in daemon mode\n");
+    mprintf("    --foreground         -f              run daemon in foreground\n");
     mprintf("    --pid=FILE           -p FILE         save daemon's pid in FILE\n");
     mprintf("    --user=USER          -u USER         run as USER\n");
     mprintf("    --no-dns                             force old non-DNS verification method\n");
@@ -418,7 +419,7 @@
 	bigsleep = 24 * 3600 / checks;
 
 #if !defined(C_OS2) && !defined(C_WINDOWS)
-	if(!optget(opts, "Foreground")->enabled) {
+	if(!optget(opts, "Foreground")->enabled && !optget(opts, "foreground")) {
 	    if(daemonize() == -1) {
 		logg("!daemonize() failed\n");
 		optfree(opts);
diff -Naur Clamav-0.95.2/shared/optparser.c clamav-0.95.2/shared/optparser.c
--- Clamav-0.95.2/shared/optparser.c	2009-05-21 17:29:08.000000000 +0530
+++ clamav-0.95.2/shared/optparser.c	2009-06-15 14:31:19.000000000 +0530
@@ -69,6 +69,7 @@
     { NULL, "version", 'V', TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMD | OPT_FRESHCLAM | OPT_CLAMSCAN | OPT_CLAMDSCAN | OPT_SIGTOOL | OPT_MILTER | OPT_CLAMCONF | OPT_CLAMDTOP, "", "" },
     { NULL, "debug", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMD | OPT_FRESHCLAM | OPT_CLAMSCAN | OPT_SIGTOOL, "", "" },
     { NULL, "verbose", 'v', TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_FRESHCLAM | OPT_CLAMSCAN | OPT_CLAMDSCAN | OPT_SIGTOOL, "", "" },
+    { NULL, "foreground", 'f', TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_FRESHCLAM | OPT_CLAMSCAN | OPT_CLAMDSCAN | OPT_SIGTOOL, "", "" },
     { NULL, "quiet", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_FRESHCLAM | OPT_CLAMSCAN | OPT_CLAMDSCAN | OPT_SIGTOOL, "", "" },
     { NULL, "leave-temps", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_CLAMSCAN, "", "" },
     { NULL, "no-warnings", 0, TYPE_BOOL, MATCH_BOOL, 0, NULL, 0, OPT_FRESHCLAM, "", "" },
